project/cyk.py

```diff
@@ -0,0 +1,46 @@
+from pyformlang.cfg import CFG
+
+__all__ = ["cyk"]
+
+
+def cyk(cfg: CFG, word: str) -> bool:
+    """
+    Determines whether the given string can be generate in the given context-free grammar
+
+    Parameters
+    ---------
+    cfg: CFG
+        Given context-free grammar
+    word: str
+        Given word
+
+    Returns
+    -------
+    bool
+        If the given string can be generate in the given context-free grammar
+    """
+    word_len = len(word)
+
+    if word_len == 0:
+        return cfg.generate_epsilon()
+
+    cnf = cfg.to_normal_form()
+
+    d = [[set() for _ in range(word_len)]] * word_len
```

Comment:
```python
>>> d = [[set() for _ in range(2)]] * 2
>>> d
[[set(), set()], [set(), set()]]
>>> d[0][0].add(1)
>>> d
[[{1}, set()], [{1}, set()]]
```
Я думаю, вы такого не ожидали.

Лучше сделать так.
```suggestion
    d = [[set() for _ in range(word_len)] for _ in range(word_len)]
```

Тогда будет
```python
>>> d = [[set() for _ in range(2)] for _ in range(2)] 
>>> d
[[set(), set()], [set(), set()]]
>>> d[0][0].add(1)
>>> d
[[{1}, set()], [set(), set()]]
```